components/email_aliases/email_aliases_service.cc

```diff
@@ -30,21 +90,126 @@ void EmailAliasesService::BindInterface(
   receivers_.Add(this, std::move(receiver));
 }
 
+void EmailAliasesService::NotifyObserversAuthStateChanged(
+    mojom::AuthenticationStatus status) {
+  for (auto& observer : observers_) {
+    observer->OnAuthStateChanged(
+        mojom::AuthState::New(status, auth_email_, std::nullopt));
+  }
+}
+
+void EmailAliasesService::ApiFetch(
```

Comment:
Can we use [api_request_helper](https://github.com/brave/brave-core/blob/39ad6bc807445a250fa06f7a94855260dcad102f/components/api_request_helper/api_request_helper.h#L101) here so that the responses from the server don't accidentally introduce crashes or other issues.

It's an abstraction meant to help with rule of 2 adherence.